ARE/services/WebService/src/main/java/eu/asterics/mw/webservice/serverUtils/ServerRepository.java

```diff
@@ -35,17 +39,98 @@
  *
  */
 public class ServerRepository {
+    //Singleton instance
+    private static ServerRepository instance;
+    
+	
 
     // Rest Server configuration
+    public static final String ARE_WEBSERVICE_PORT_REST_KEY = "ARE.webservice.port.REST";
     public static final String PATH_REST = "/rest";
-    public static final int PORT_REST = 8081;
-    public static final URI BASE_URI_REST = URI.create("http://0.0.0.0:" + PORT_REST + PATH_REST);
-
-    // Web Service Server configuration
-    public static final String PATH_WS = "/ws";
-    public static final String PATH_WS_ASTERICS_DATA = "/astericsData";
-    public static final int PORT_WS = 8082;
-    public static final URI BASE_URI_WS = URI.create("http://localhost:" + PORT_WS + PATH_WS);
+    public static final int DEFAULT_PORT_REST = 8081;    
+    
+    // Web Socket Server configuration
+    public static final String PATH_WEBSOCKET = "/ws";
+    public static final String PATH_WEBSOCKET_ASTERICS_DATA = "/astericsData";
+    public static final String ARE_WEBSERVICE_PORT_WEBSOCKET_KEY = "ARE.webservice.port.websocket";    
+    public static final int DEFAULT_PORT_WEBSOCKET = 8082;
+    
+    //member variables holding property values
+    private int portREST=DEFAULT_PORT_REST;
+    private int portWebsocket=DEFAULT_PORT_WEBSOCKET;
+    
+    /**
+     * Private ctor used for initializing the class.
+     */
+    private ServerRepository() {
+    	//init ports and paths with property values
+    	try {
+    		portREST=Integer.parseInt(AREProperties.instance.getProperty(ARE_WEBSERVICE_PORT_REST_KEY, String.valueOf(DEFAULT_PORT_REST)));
+    		//We have to store back the actual property value so that it will be written to the areProperties config file if it did not exist before.
+    		AREProperties.instance.setProperty(ARE_WEBSERVICE_PORT_REST_KEY, String.valueOf(portREST));
```

Comment:
why do we have to store back the value? why it is not ok, if the property does not exist in the property-file?

Reply:
ah, ok, i think this just sets the property internally on the property instance, so that further calls to get this property return the correct value. correct?